File: app/src/main/java/com/example/todolistapp/TodoPagerActivity.java
package com.example.todolistapp;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import androidx.viewpager2.widget.ViewPager2;


import java.util.List;
import java.util.UUID;

/**
 * Created by Ebbi on 26/11/2017.
 * Updated by Patrick for Android 12 18/1/2022
 */

public class TodoPagerActivity extends AppCompatActivity {

    private static final String EXTRA_TODO_ID = "todo_id";

    private List<Todo> mTodos;

    public static Intent newIntent(Context packageContext, UUID todoId){
        Intent intent = new Intent(packageContext, TodoPagerActivity.class);
        intent.putExtra(EXTRA_TODO_ID, todoId);
        return intent;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_todo_pager);

        UUID todoId = (UUID) getIntent().getSerializableExtra(EXTRA_TODO_ID);

        ViewPager2 mViewPager = findViewById(R.id.todo_view_pager);

        mTodos = TodoModel.get(this).getTodos();

       mViewPager.setAdapter(new FragmentStateAdapter(this) {
            @Override
            public int getItemCount() {
                return mTodos.size();
            }

            @NonNull
            @Override
            public Fragment createFragment(int position) {
                Todo todo = mTodos.get(position);
                return TodoFragment.newInstance(todo.getId());
            }


        });

        for (int i = 0; i < mTodos.size(); i++){
            if (mTodos.get(i).getId().equals(todoId)) {
                mViewPager.setCurrentItem(i);
                break;
            }
        }

    }
}

File: app/src/main/java/com/example/todolistapp/TodoPagerActivity.kt
package com.example.todolistapp

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.ui.Modifier
import com.example.todolistapp.data.TodoModel
import com.example.todolistapp.model.Todo
import com.example.todolistapp.ui.TodoSwiper
import com.example.todolistapp.ui.theme.TodoTheme
import java.util.UUID

/**
 * Updated by Patrick for Android 14 Compose 29/2/2024
 */

class TodoPagerActivity : ComponentActivity() {
    private var mTodos: List<Todo>? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val todoId = intent.getSerializableExtra(EXTRA_TODO_ID, UUID::class.java)
        mTodos = TodoModel.Companion.get()?.todos
        val currentPage:Int = mTodos!!.indexOfFirst { it.id == todoId }

        setContent {
            TodoTheme {
                // A surface container using the 'background' color from the theme
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = MaterialTheme.colorScheme.background
                ) {
                    TodoSwiper(mTodos!!, currentPage, Modifier)
                }
            }
        }
    }

    companion object {
        private const val EXTRA_TODO_ID = "todo_id"
        fun newIntent(packageContext: Context?, todoId: UUID?): Intent {
            val intent = Intent(packageContext, TodoPagerActivity::class.java)
            intent.putExtra(EXTRA_TODO_ID, todoId)
            return intent
        }
    }
}